modules/fundamental/src/sbom/endpoints/mod.rs

```diff
@@ -70,11 +71,44 @@ pub fn configure(
         .service(download)
         .service(label::set)
         .service(label::update)
+        .service(get_unique_licenses)
         .service(get_license_export);
 }
 
 const CONTENT_TYPE_GZIP: &str = "application/gzip";
 
+#[utoipa::path(
+    tag = "sbom",
+    operation_id = "getUniqueLicenses",
+    params(
+    ("id" = Id, Path),
+    ),
+    responses(
+    (status = 200, description = "fetch all unique license id and license info id", body = Vec<String>),
+    (status = 404, description = "not found"),
+    ),
+)]
+#[get("/v2/sbom/{id}/licenseids")]
+pub async fn get_unique_licenses(
+    fetcher: web::Data<LicenseService>,
+    db: web::Data<Database>,
+    id: web::Path<String>,
+    _: Require<ReadSbom>,
+) -> actix_web::Result<impl Responder> {
+    let id = Id::from_str(&id).map_err(Error::IdKey)?;
+    let sbom_id = if let Some(uuid) = id.try_as_uid() {
+        uuid
+    } else {
+        Uuid::new_v4()
+    };
+    let spdx_licenses = fetcher.get_unique_licenses(sbom_id, db.as_ref()).await?;
```

Comment:
As this should return a consistent set, I think we this should also use the same database transaction.